src/modules/utils/buffer.ts

```diff
@@ -15,38 +15,85 @@
 
 *******************************************************************************/
 
+import * as assert from "assert";
+
+/**
+ * The byte order
+ */
+export enum Endian
+{
+    Little,
+    Big
+}
+
 /**
  * Read from the hex string
  * @param hex The hex string
  * @param target The buffer to output
+ * @param endian The byte order
  * @returns The output buffer
  */
-export function readFromString (hex: string, target?: Buffer): Buffer
+export function readFromString (hex: string, target?: Buffer, endian?: Endian): Buffer
 {
     let start = (hex.substr(0, 2) == '0x') ? 2 : 0;
     let length = (hex.length - start) >> 1;
-
     if (target == undefined)
         target = Buffer.alloc(length);
 
-    for (let pos = 0, idx = start; idx < length * 2 + start; idx += 2, pos++)
-        target[length - pos - 1] = parseInt(hex.substr(idx, 2), 16);
-
+    if ((endian === undefined) || (endian == Endian.Little))
+    {
+        for (let pos = 0, idx = start; idx < length * 2 + start; idx += 2, pos++)
+            target[length - pos - 1] = parseInt(hex.substr(idx, 2), 16);
+    }
+    else
+    {
+        for (let pos = 0, idx = start; idx < length * 2 + start; idx += 2, pos++)
+            target[pos] = parseInt(hex.substr(idx, 2), 16);
+    }
     return target;
 }
 
 /**
  * Write to the hex string
  * @param source The buffer to input
+ * @param endian The byte order
  * @returns The hex string
  */
-export function writeToString (source: Buffer): string
+export function writeToString (source: Buffer, endian?: Endian): string
 {
-    let hex = [];
-    for (let idx = source.length-1; idx >= 0; idx--)
+    if ((endian === undefined) || (endian == Endian.Little))
     {
-        hex.push((source[idx] >>> 4).toString(16));
-        hex.push((source[idx] & 0xF).toString(16));
+        let hex = [];
+        for (let idx = source.length-1; idx >= 0; idx--) {
```

Comment:
I think there should be an assert or check here somewhere, what if the length is zero?